From 8f8c830c36c6839acb9d570189f64b42853ed02b Mon Sep 17 00:00:00 2001
From: Renato Araujo Oliveira Filho <renato.filho@canonical.com>
Date: Mon, 30 Jun 2014 11:16:19 -0300
Subject: [PATCH] Fixed DisplayName conversion to vcard.

Based on VCARD spec "FN" field is only one string, the old implementation store a string list
and does not change the property type, that causes the following error:

Variant value is a QStringList but the property's value type is neither CompoundType or ListType

Changing the property to store a single string fixes the problem.

Change-Id: I7a7857fb2c498518d7940410d0bb65558e230fec
Reviewed-by: Matthew Vogt <matthew.vogt@qinetic.com.au>
Reviewed-by: Christopher Adams <chris.adams@jollamobile.com>
---
 src/versit/qvcard30writer.cpp           | 4 ++--
 src/versit/qversitcontactexporter_p.cpp | 4 +---
 2 files changed, 3 insertions(+), 5 deletions(-)

diff --git a/src/versit/qvcard30writer.cpp b/src/versit/qvcard30writer.cpp
index 924ef69..20d01aa 100644
--- a/src/versit/qvcard30writer.cpp
+++ b/src/versit/qvcard30writer.cpp
@@ -108,8 +108,8 @@ void QVCard30Writer::encodeVersitProperty(const QVersitProperty& property)
             separator = QStringLiteral(";");
         } else {
             if (property.valueType() != QVersitProperty::ListType) {
-                qWarning("Variant value is a QStringList but the property's value type is neither "
-                         "CompoundType or ListType");
+                qWarning("Property: %s, variant value is a QStringList but the property's value type is neither "
+                         "CompoundType or ListType", property.name().toLatin1().constData());
             }
             // Assume it's a ListType
             separator = QStringLiteral(",");
diff --git a/src/versit/qversitcontactexporter_p.cpp b/src/versit/qversitcontactexporter_p.cpp
index 85458d6..6b30126 100644
--- a/src/versit/qversitcontactexporter_p.cpp
+++ b/src/versit/qversitcontactexporter_p.cpp
@@ -526,9 +526,7 @@ void QVersitContactExporterPrivate::encodeDisplayLabel(
     QVersitProperty property =
         VersitUtils::takeProperty(document, QStringLiteral("FN"), removedProperties);
     property.setName(QStringLiteral("FN"));
-    QStringList value(property.variantValue().toStringList());
-    value.append(displaylabelDetail.label());
-    property.setValue(value);
+    property.setValue(displaylabelDetail.label());
     *generatedProperties << property;
     *processedFields << QContactDisplayLabel::FieldLabel;
 }
-- 
2.5.0

